documentation/src/main/asciidoc/topics/con_expiration_overview.adoc

```diff
@@ -0,0 +1,38 @@
+[id='how_expiration_works-{context}']
+= How Expiration Works
+By default, `Cache.put()` creates immortal entries. With expiration, `Cache.put()` creates mortal entries with lifespans and maximum idle times.
+
+When you configure expiration, {brandname} stores keys with metadata that
+determines when entries expire.
+
+* Lifespan uses a `creation` timestamp and the value for the `lifespan` configuration property.
+* Maximum idle uses a `last used` timestamp and the value for the `max-idle` configuration property.
+
+{brandname} checks if lifespan or maximum idle metadata is set and then
+compares the values with the current time.
+
+If `(creation + lifespan > currentTime)` or `(lastUsed + maxIdle > currentTime)` then {brandname} detects that the entry is expired.
+
+Expiration occurs whenever entries are accessed or iterated on.
+
+For example, `k1` reaches the maximum idle time and a client makes a
+`Cache.get(k1)` request. In this case, {brandname} detects that the entry is
+expired and removes it. The `Cache.get()` call does not return.
+
+//dnaro: contradicting info here. does this apply with local cache stores only?
+{brandname} first removes entries from the data container and then from the
+cache store, if configured.
+
+[NOTE]
+====
+{brandname} adds expiration metadata as `long` primitive data types to cache entries.
+
+If you configure lifespan expiration, {brandname} adds only the corresponding
+metadata and likewise for maximum idle. If you configure both lifespan and
+maximum idle expiration, {brandname} adds all metadata to keys.
+
+`long` primitive data types are eight bytes in size. If you configure either
```

Comment:
I am not sure if we need to go into this much detail about the sizes of the objects and such here.